lib/prelude.dx

```diff
@@ -805,12 +851,73 @@ Dex's PRNG system is modelled directly after [JAX's](https://github.com/google/j
 '### Key functions
 
 -- TODO: newtype
-Key = Int64
-
-def hash (x:Key) (y:Int32) : Key = unsafeIO do
-  y64 = IToI64 y
-  %ffi threefry2x32 Int64 x y64
-def newKey (x:Int) : Key = hash (IToI64 0) x
+Key = Word64
+
+def threeFry2x32C (k:Int64) (count:Int64) : Int64 = unsafeIO do
+  %ffi threefry2x32 Int64 k count
+
+def applyRound (x:Word32) (y:Word32) (rot:Int) : (Word32&Word32) =
+  x = x + y
+  y = (y << rot) .|. (y >> (32 - rot))
+  y = x ^ y
+  (x, y)
+
+def threeFry2x32Dex (k:Word64) (count:Word64) : Word64 = 
+  -- Based on jax's threefry_2x32 by Matt Johnson and Peter Hawkins
+  rotations1 = [13, 15, 26, 6]
+  rotations2 = [17, 29, 16, 24]
+
+  k0 = W64ToW32 (k >> 32)
+  k1 = W64ToW32 k
```

Comment:
I kind of dislike `W64ToW32` because its semantics are a big ambiguous (does it slice of the low bits? high bits? something else?). How about we add functions
```
highWord : Word64 -> Word32
lowWord : Word64 -> Word32
```
and use that to do the projections instead.